app/repositories/slick/implementations/SlickChatsRepository.scala

```diff
@@ -259,24 +286,289 @@ class SlickChatsRepository @Inject() (db: Database)(implicit executionContext: E
+    for {
+      addresses <- EmailAddressesTable.all.join(AddressesTable.all)
+        .on((emailAddressRow, addressRow) => emailAddressRow.emailId === emailId && emailAddressRow.addressId === addressRow.addressId)
+        .map {
+          case (emailAddressRow, addressRow) =>
+            (emailAddressRow.participantType, addressRow.addressId, addressRow.address)
+        }
+        .result
+    } yield addresses.groupBy(_._1) //groupBy participantType
+  }
+
+  /**
+   * Method that gets the from address of an email. It also verifies if this from address is the user's address
+   * and if the email (with given emailId) is a part of the chat (with given chatId)
+   * @param chatId ID of the chat
+   * @param emailId ID of the email
+   * @param userId ID of the user of the address to return
+   * @return address of the user with userId that is also the sender ("from") of the email
+   */
+  private def getVerifiedFromAddressQuery(chatId: String, emailId: String, userId: String): Query[Rep[String], String, scala.Seq] = {
+    UserChatsTable.all
+      .join(EmailsTable.all).on {
+        case (userChatRow, emailRow) => userChatRow.chatId === emailRow.chatId &&
+          userChatRow.chatId === chatId && userChatRow.userId === userId &&
+          userChatRow.draft > 0 && emailRow.emailId === emailId && emailRow.sent === 0 //must be a draft from this user
+      }.map(_._2)
+      .join(EmailAddressesTable.all).on {
+        case (emailRow, emailAddressRow) => emailRow.emailId === emailAddressRow.emailId &&
+          emailAddressRow.participantType === "from"
+      }.map(_._2)
+      .join(AddressesTable.all).on {
+        case (emailAddressRow, addressRow) => emailAddressRow.addressId === addressRow.addressId
+      }.map(_._2)
+      .join(UsersTable.all).on {
+        case (addressRow, userRow) => userRow.userId === userId &&
+          addressRow.addressId === userRow.addressId
+      }.map { case (addressRow, userRow) => addressRow.address }
+  }
+
+  /**
+   * Method that, given the receiver participation type (to, bcc, cc):
+   * - Inserts new email addresses in the database if the patch contains new ones
+   * - Deletes old email addresses from the database if the patch does not include them
+   * @param emailId ID of the email
+   * @param chatId ID of the chat
+   * @param participantType type of participant (to, bcc or cc)
+   * @param optionNewAddresses optional set of new addresses of the referred participantType. If it is None,
+   *                           no addresses will be added or deleted
+   * @return an optional set of the addresses that stayed in the database after all the additions and deletions
+   */
+  private def insertAndDeleteAddressesByParticipantType(emailId: String, chatId: String,
+    participantType: String, optionNewAddresses: Option[Set[String]]) = {
+
```

Comment:
Please make it clear that this method returns an Action, either by changing it's name, specifying it's return type, or pointing it out in it's description.